modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java

```diff
@@ -67,15 +70,35 @@ private void startServices() throws MuleException {
     }
   }
 
+  /**
+   * Stop all discovered services. In the case of 'HTTP' and 'Scheduler' service, if present, they should be stopped last and in
+   * that order.
+   *
+   * @throws MuleException
+   */
   @Override
   public void stop() throws MuleException {
+    LinkedList<Service> servicesWithSpecialStoppingOrder = new LinkedList<>();
     for (Service service : services) {
-      try {
-        stopIfNeeded(service);
-      } catch (Exception e) {
-        LOGGER.warn(format("Failed to stop service '%s': %s", service.getName(), e.getMessage()), e);
+      if (service.getName().equals(SCHEDULER_SERVICE_ARTIFACT_ID)) {
+        servicesWithSpecialStoppingOrder.addLast(service);
+      } else if (service.getName().contains(HTTP_SERVICE_ARTIFACT_PREFIX)) {
+        servicesWithSpecialStoppingOrder.addFirst(service);
+      } else {
+        doStopService(service);
       }
     }
+    for (Service service : servicesWithSpecialStoppingOrder) {
```

Comment:
you can leave it as it is if you want.. Keep in mind that you could have used `servicesWithSpecialStoppingOrder.forEach(this::doStopService)` instead